zone/tradeskills.cpp

```diff
@@ -810,6 +810,14 @@ void Client::SendTradeskillSearchResults(
 				continue;
 			}
 		}
+		
+		//Check if we need to learn it before sending them the recipe.. 
+		DBTradeskillRecipe_Struct spec;
+		if (content_db.GetTradeRecipe(recipe_id, objtype, someid, this->CharacterID(), &spec)) {
```

Comment:
Doing a select query inside of a loop like this is not ideal. We should fetch recipes once and then loop through it in memory to perform this check

Reply:
Righto I see a better way to fix it.. but looking at it.. I see other issues